query-engine/connectors/sql-query-connector/src/database/operations/read.rs

```diff
@@ -23,29 +24,59 @@ pub async fn get_single_record(
         Err(e) => Err(e),
     })?
     .map(Record::from)
-    .map(|record| SingleRecord { record, field_names });
+    .map(|record| SingleRecord {
+        record,
+        field_names,
+    });
 
     Ok(record)
 }
 
 pub async fn get_many_records(
     conn: &dyn QueryExt,
     model: &ModelRef,
-    query_arguments: QueryArguments,
+    mut query_arguments: QueryArguments,
     selected_fields: &SelectedFields,
 ) -> crate::Result<ManyRecords> {
     let field_names = selected_fields.db_names().map(String::from).collect();
     let idents: Vec<_> = selected_fields.types().collect();
-    let query = read::get_records(model, selected_fields.columns(), query_arguments);
-
-    let records = conn
-        .filter(query.into(), idents.as_slice())
-        .await?
-        .into_iter()
-        .map(Record::from)
-        .collect();
-
-    Ok(ManyRecords { records, field_names })
+    let mut records = ManyRecords::new(field_names);
+
+    if query_arguments.can_batch() {
+        // We don't need to order in the database due to us ordering in this
+        // function.
+        let order = query_arguments.order_by.take();
+
+        let batches = query_arguments.batched();
+        let mut futures = FuturesUnordered::new();
+
+        for args in batches.into_iter() {
+            let query = read::get_records(model, selected_fields.columns(), args);
+            futures.push(conn.filter(query.into(), idents.as_slice()));
+        }
+
+        while let Some(result) = futures.next().await {
+            for item in result?.into_iter() {
+                records.push(Record::from(item))
+            }
+        }
+
+        if let Some(ref order_by) = order {
+            records.order_by(order_by)
```

Comment:
Hmm so here we actually sort ourselves right?

Reply:
yep